rust/azure_iot_operations_services/src/azure_device_registry.rs

```diff
@@ -61,6 +63,9 @@ pub enum ErrorKind {
     /// An error occurred while shutting down the Azure Device Registry Client.
     #[error("Shutdown error occurred with the following protocol errors: {0:?}")]
     ShutdownError(Vec<AIOProtocolError>),
+    /// An error occured while validating the inputs.
+    #[error("{0} must not be empty")]
+    ValidationError(String),
```

Comment:
Is this new error type really necessary? Couldn't we make just a single kind that is for all validation issues?

Reply:
This is the only one for validation issue right now, if we have other validation issues in the future we can always combine

Reply:
merged the invalid client id and validation one